src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/DataContract.cs

```diff
@@ -1726,6 +1812,59 @@ internal static string GetDefaultStableNamespace(string clrNs)
             return new Uri(Globals.DataContractXsdBaseNamespaceUri, clrNs).AbsoluteUri;
         }
 
+        internal static void GetDefaultStableName(string fullTypeName, out string localName, out string ns)
+        {
+            CodeTypeReference typeReference = new CodeTypeReference(fullTypeName);
+            GetDefaultStableName(typeReference, out localName, out ns);
+        }
+
+        private static void GetDefaultStableName(CodeTypeReference typeReference, out string localName, out string ns)
+        {
+            string fullTypeName = typeReference.BaseType;
+            DataContract dataContract = GetBuiltInDataContract(fullTypeName);
+            if (dataContract != null)
+            {
+                localName = dataContract.StableName.Name;
+                ns = dataContract.StableName.Namespace;
+                return;
+            }
+
+            GetClrNameAndNamespace(fullTypeName, out localName, out ns);
+            if (typeReference.TypeArguments.Count > 0)
+            {
+                StringBuilder localNameBuilder = new StringBuilder();
+                StringBuilder argNamespacesBuilder = new StringBuilder();
+                bool parametersFromBuiltInNamespaces = true;
+                IList<int> nestedParamCounts = GetDataContractNameForGenericName(localName, localNameBuilder);
```

Comment:
I realize there are a bunch of allocations going on in this method, but one allocation could be avoided by changing the return type of `GetDataContractNameForGenericName` from `IList<int>` to `List<int>` and `nestedParamCounts` to `List<int>`, which would avoid the enumerator allocation in the `foreach` below (since it would use `List<T>`'s struct enumerator instead of the boxed enumerator that would be returned from `IList<int>`).